### PR TITLE
fix: add cleanup of test vm in script

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind fix

**What this PR does / why we need it**:
add cleanup of test vm in script.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: .pipelines/templates/.builder-release-template-windows.yaml [hunk 1]
```diff
@@ -218,6 +218,7 @@ steps:
       -e DRY_RUN=${DRY_RUN} \
       -e IMAGE_NAME=${IMAGE_NAME} \
       -e VNET_RESOURCE_GROUP_NAME=${VNET_RESOURCE_GROUP_NAME} \
+      -e TEST_VM_RESOURCE_GROUP_NAME=${TEST_VM_RESOURCE_GROUP_NAME} \
```
Comment (AbelHu) [inline .pipelines/templates/.builder-release-template-windows.yaml:221]: How is`TEST_VM_RESOURCE_GROUP_NAME` generated in `vhdbuilder/packer/test/run-test.sh` passed here?

File: vhdbuilder/packer/test/run-test.sh [hunk 2]
```diff
@@ -25,16 +25,17 @@ if [ "${OS_TYPE,,}" == "linux" ]; then
   AZURE_LOCATION=$PACKER_BUILD_LOCATION
 fi
 
-RESOURCE_GROUP_NAME="$TEST_RESOURCE_PREFIX-$(date +%s)-$RANDOM"
-az group create --name $RESOURCE_GROUP_NAME --location ${AZURE_LOCATION} --tags 'source=AgentBaker'
+TEST_VM_RESOURCE_GROUP_NAME="$TEST_RESOURCE_PREFIX-$(date +%s)-$RANDOM"
```
Comment (AbelHu) [inline vhdbuilder/packer/test/run-test.sh:28]: Remove this line and check whether the variable is passed?

Comment (junjiezhang1997) [reply vhdbuilder/packer/test/run-test.sh:28]: The local code was not saved and uploaded.. Will update it.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**